File: backend/app/api/v1/routes/hr_cockpit.py
from uuid import UUID

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.authz.deps import get_current_employee, require_permission
from app.authz.enums import Action, FeatureKey
from app.database import get_db
from app.models.employee import Employee
from app.schemas.hr_content import HolidayContentIn, JobDescriptionIn, JobOpeningIn, LearningEventIn, PolicyIn
from app.services import hr_cockpit_service as service

router = APIRouter(prefix="/hr-cockpit", tags=["hr-cockpit"])


@router.get("/dashboard")
async def dashboard(db: AsyncSession = Depends(get_db), actor: Employee = Depends(require_permission(FeatureKey.HR_COCKPIT, Action.VIEW))):
    return await service.dashboard(db, actor)


@router.post("/policies", status_code=201)
async def create_policy(payload: PolicyIn, db: AsyncSession = Depends(get_db), actor: Employee = Depends(get_current_employee)):
    return await service.create_policy(db, actor, payload)


@router.post("/learning-events", status_code=201)
async def create_event(payload: LearningEventIn, db: AsyncSession = Depends(get_db), actor: Employee = Depends(get_current_employee)):
    return await service.create_event(db, actor, payload)


@router.post("/job-descriptions", status_code=201)
async def create_jd(payload: JobDescriptionIn, db: AsyncSession = Depends(get_db), actor: Employee = Depends(get_current_employee)):
    return await service.create_jd(db, actor, payload)


@router.post("/openings", status_code=201)
async def create_opening(payload: JobOpeningIn, db: AsyncSession = Depends(get_db), actor: Employee = Depends(get_current_employee)):
    return await service.create_opening(db, actor, payload)


@router.post("/holidays", status_code=201)
async def create_holiday(payload: HolidayContentIn, db: AsyncSession = Depends(get_db), actor: Employee = Depends(get_current_employee)):
    return await service.create_holiday(db, actor, payload)


@router.post("/{kind}/{item_id}/submit")
async def submit(kind: str, item_id: UUID, db: AsyncSession = Depends(get_db), actor: Employee = Depends(get_current_employee)):
    return await service.submit_content(db, actor, kind, item_id)


@router.post("/{kind}/{item_id}/{decision}")
async def decide(kind: str, item_id: UUID, decision: str, db: AsyncSession = Depends(get_db), actor: Employee = Depends(get_current_employee)):
    return await service.approve_or_publish(db, actor, kind, item_id, decision)

File: frontend/src/api/hrCockpit.ts
import { apiClient } from "./client";

export type ContentItem = Record<string, any> & { id: string; status: string };
export type HrDashboard = { role: string; policies: ContentItem[]; events: ContentItem[]; job_descriptions: ContentItem[]; openings: ContentItem[]; referrals: ContentItem[]; holidays: ContentItem[]; hiring_managers: {id:string;name:string}[] };

export const hrCockpitApi = {
  dashboard: () => apiClient.get<HrDashboard>("/hr-cockpit/dashboard").then(r => r.data),
  createPolicy: (p: unknown) => apiClient.post("/hr-cockpit/policies", p).then(r => r.data),
  createEvent: (p: unknown) => apiClient.post("/hr-cockpit/learning-events", p).then(r => r.data),
  createJd: (p: unknown) => apiClient.post("/hr-cockpit/job-descriptions", p).then(r => r.data),
  createOpening: (p: unknown) => apiClient.post("/hr-cockpit/openings", p).then(r => r.data),
  createHoliday: (p: unknown) => apiClient.post("/hr-cockpit/holidays", p).then(r => r.data),
  transition: (kind: string, id: string, action: string) => apiClient.post(`/hr-cockpit/${kind}/${id}/${action}`).then(r => r.data),
};

export const publicContentApi = {
  policies: () => apiClient.get<ContentItem[]>("/public/content/policies").then(r => r.data),
  events: () => apiClient.get<ContentItem[]>("/public/content/learning-events").then(r => r.data),
  openings: () => apiClient.get<ContentItem[]>("/public/content/openings").then(r => r.data),
  holidays: () => apiClient.get<ContentItem[]>("/public/content/holidays").then(r => r.data),
  refer: (id: string, p: unknown) => apiClient.post(`/public/content/openings/${id}/referrals`, p).then(r => r.data),
};
